examples/vision/object_detection_using_vision_transformer.py

```diff
@@ -0,0 +1,605 @@
+ load the TensorBoard notebook extension inside notebook
+```
+%load_ext tensorboard
+```
+"""
+
+"""
+## Prepare dataset
+"""
+
+# Dataset
+# http://www.vision.caltech.edu/Image_Datasets/Caltech101/
+# Download the images '101_ObjectCategories.tar.gz' and
+# annotations 'Annotations.tar' available seperately
+# Extract images (using airplanes here) to images folder and
+# all annotations to another folder.
+
+# Path to images and annotations
+PATH_IMAGES = "C:/Users/DAVE/Desktop/Object detection with transformers/airplanes/"
+PATH_ANNOT = (
+    "C:/Users/DAVE/Desktop/Object detection with transformers/Airplanes_Side_2/"
+)
+
+# list of paths to images and annotations
+image_paths = [
+    f for f in os.listdir(PATH_IMAGES) if os.path.isfile(os.path.join(PATH_IMAGES, f))
+]
+annot_paths = [
+    f for f in os.listdir(PATH_ANNOT) if os.path.isfile(os.path.join(PATH_ANNOT, f))
+]
+
+annot_paths.sort()
+image_paths.sort()
+
+images = []
+targets = []
+
+# loop over the annotation rows, preprocess and store in list
+for i in range(0, len(annot_paths)):
+    #   Access bounding box coordinates
+    annot = scipy.io.loadmat(PATH_ANNOT + annot_paths[i])["box_coord"][0]
+
+    topLeft_X = annot[2]
+    topLeft_Y = annot[0]
+    bottomRight_X = annot[3]
+    bottomRight_Y = annot[1]
+
+    image = cv2.imread(
+        "C:/Users/DAVE/Desktop/Object detection with transformers/airplanes/"
+        + image_paths[i]
+    )
+    (h, w) = image.shape[:2]
+
+    # apply relative scaling to bounding boxes as per given image
+    topLeft_X = float(topLeft_X) / w
+    topLeft_Y = float(topLeft_Y) / h
+    bottomRight_X = float(bottomRight_X) / w
+    bottomRight_Y = float(bottomRight_Y) / h
+
+    # load the image and convert to array. Store image to images list
+    # and bounding box in targets list
+    image = load_img(
+        "C:/Users/DAVE/Desktop/Object detection with transformers/airplanes/"
+        + image_paths[i],
+        target_size=(224, 224),
+    )
+    image = img_to_array(image)
+    images.append(image)
+    targets.append((topLeft_X, topLeft_Y, bottomRight_X, bottomRight_Y))
+
+# Print Shape
+# print(np.shape(images))
+# print(np.shape(targets))
+
+# Convert to numpy and split train and test dataset
+(x_train) = np.asarray(images[: int(len(images) * 0.8)])
+(y_train) = np.asarray(targets[: int(len(targets) * 0.8)])
+# (x_test) = np.asarray(images[int(len(images) * 0.8) :])
+# (y_test) = np.asarray(targets[int(len(targets) * 0.8) :])
+
+# print(f"x_train shape: {x_train.shape} - y_train shape: {y_train.shape}")
+# print(f"x_test shape: {x_test.shape} - y_test shape: {y_test.shape}")
+
+"""
+## Data augmentation
+"""
+
+# Some Code referred from
+# https://keras.io/examples/vision/image_classification_with_vision_transformer/
+
+image_size = 64  # resize input images to this size
+patch_size = 8  # Size of the patches to be extracted from the input images
+
+data_augmentation = keras.Sequential(
+    [
+        layers.Normalization(),
```

Comment:
Normalization or rescaling should happen as part of the model, not as part of the augmentation stage.